x11/xauth.go

```diff
@@ -0,0 +1,133 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package x11
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+)
+
+type xauth struct {
+	Family  uint16
+	Address []byte
+	Number  []byte
+	Name    []byte
+	Data    []byte
+}
+
+func readChunk(f *os.File) ([]byte, error) {
+	b := make([]byte, 2)
+	n, err := f.Read(b)
+	if err != nil {
+		return nil, err
+	} else if n != 2 {
```

Comment:
Sorry, I did not express myself very well. I was thinking of just a tiny style change:
```
n, err := f.Read(b)
+	if err != nil {
+		return nil, err
+	} 
+       if n != 2 {
```
(i.e. drop the "else" but keep the check for n!=2)